kimchi/benches/proof_criterion_mina.rs

```diff
@@ -0,0 +1,97 @@
+#![allow(clippy::unit_arg)]
+use criterion::{black_box, criterion_group, criterion_main, Criterion, SamplingMode};
+use groupmap::GroupMap;
+use kimchi::{
+    bench::{
+        bench_arguments_from_file, BaseSpongePallas, BaseSpongeVesta, ScalarSpongePallas,
+        ScalarSpongeVesta,
+    },
+    curve::KimchiCurve,
+    proof::ProverProof,
+};
+use mina_curves::pasta::{Pallas, Vesta};
+
+pub fn bench_proof_creation_mina(c: &mut Criterion) {
+    let mut group = c.benchmark_group("proof_creation_mina");
+    // we don't have to be precise with these benches
+    group.sampling_mode(SamplingMode::Flat); // for slow benchmarks
+
+    group.sample_size(10); // Limits the number of samples
+
+    let filename = std::env::var("BENCH_PROOF_CREATION_MINA_INPUTS").unwrap();
+
+    // Parse filename "kimchi_inputs_CURVENAME_SEED.ser" into two parameters
+    let (curve_name, seed): (&str, &str) = filename
+        .split('/')
+        .last()
+        .unwrap()
+        .strip_prefix("kimchi_inputs_")
+        .unwrap()
+        .strip_suffix(".ser")
+        .unwrap()
+        .split_once('_')
+        .unwrap();
+
+    if curve_name == Vesta::NAME {
+        // Vesta
+        let srs = kimchi::precomputed_srs::get_srs_test();
+        let (index, witness, runtime_tables, prev) =
+            bench_arguments_from_file::<Vesta, BaseSpongeVesta>(srs.clone(), filename.clone());
+
+        let group_map = GroupMap::<_>::setup();
+        group.bench_function(
+            format!("proof creation (mina, vesta, circuit seed {})", seed),
+            |b| {
+                b.iter(|| {
+                    black_box(ProverProof::create_recursive::<
+                        BaseSpongeVesta,
+                        ScalarSpongeVesta,
+                        _,
+                    >(
+                        &group_map,
+                        witness.clone(),
```

Comment:
Recommend wrapping these `clone`s in `black_box`es, since the compiler might be able to infer it can skip them.

I also wonder if you meant for these to be cloned inside the benchmark?  It would affect the results.  It might be better to clone them upfront, if that's possible.

Reply:
You are right that these are part of the benchmark, but tbh the prover takes 3-4 seconds, so 0.1ms of cloning won't affect it much...